TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodyAccumulateHandler.java

```diff
@@ -74,10 +109,12 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                             jsonAccumulator.getTotalBytesFullyConsumed() + " for a single composite ByteBuf");
                     }
                     // skip the contents that were already parsed and included in the payload as parsed json
-                    accumulatedBody.readerIndex((int) jsonAccumulator.getTotalBytesFullyConsumed());
                     // and pass the remaining stream
+                    var jsonBodyByteLength = jsonWasInvalid ? 0 : (int) jsonAccumulator.getTotalBytesFullyConsumed();
                     capturedHttpJsonMessage.payload()
-                        .put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY, accumulatedBody);
+                        .put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY,
+                            accumulatedBody.retainedSlice(jsonBodyByteLength,
```

Comment:
nit: should the first argument be accumulatedBody.getReaderIndex + jsonBodyByteLength?

Reply:
I'm fine with this since this the accumulatedBody is controlled by this class